src/physics/initial_conditions/initial_condition_function.cpp

```diff
@@ -450,6 +450,173 @@ ::value(const dealii::Point<dim,real> &point, const unsigned int istate) const
     return soln_conservative[istate];
 }
 
+
+// ========================================================
+// 1D Sod Shock tube -- Initial Condition
+// See Chen & Shu, Entropy stable high order..., 2017, Pg. 25
+// ========================================================
+template <int dim, int nstate, typename real>
+InitialConditionFunction_SodShockTube<dim,nstate,real>
+::InitialConditionFunction_SodShockTube (
+        Parameters::AllParameters const* const param)
+        : InitialConditionFunction<dim,nstate,real>()
+{
+    // Euler object; create using dynamic_pointer_cast and the create_Physics factory
+    // This test should only be used for Euler
+    this->euler_physics = std::dynamic_pointer_cast<Physics::Euler<dim, dim + 2, double>>(
+        Physics::PhysicsFactory<dim, dim + 2, double>::create_Physics(param));
+}
+
+template <int dim, int nstate, typename real>
+inline real InitialConditionFunction_SodShockTube<dim,nstate,real>
+::value(const dealii::Point<dim,real> &point, const unsigned int istate) const
+{
+    real density = 0.0;
+    real vel = 0.0;
+    real pressure = 0.0;
+    if(point[0] < 0){
+        density = 1.0;
+        vel = 0.0;
+        pressure = 1.0;
+    }
+    else{
+        density = 0.125;
+        vel = 0.0;
+        pressure = 0.1;
+    }
+    std::array<real, nstate> soln_primitive;
+    soln_primitive[0] = density;
+    soln_primitive[1] = vel;
+    soln_primitive[2] = pressure;
```

Comment:
`if constexpr(dim==1 && nstate==(dim+2))`

Reply:
maybe also add these types of safeguards in other IC definitions

Reply:
Added safeguards to all 4 IC definitions